src/libs/actions/IOU.js

```diff
@@ -13,7 +16,94 @@ function getPreferredCurrency() {
     }, 1600);
 }
 
-// Re-enable the prefer-default-export lint when additional functions are added
+function setNewIOUReportObjects(reportIds) {
+    return API.Get({
+        returnValueList: 'reportStuff',
+        reportIDList: reportIds,
+        shouldLoadOptionalKeys: true,
+        includePinnedReports: true,
+    })
+        .then(({reports}) => _.map(reports, getSimplifiedIOUReport))
+        .then((iouReportObjects) => {
+            const reportIOUData = {};
+
+            if (iouReportObjects.length === 1) {
+                const iouReportKey = `${ONYXKEYS.COLLECTION.REPORT_IOUS}${iouReportObjects[0].reportID}`;
+                return Onyx.merge(iouReportKey,
+                    getSimplifiedIOUReport(iouReportObjects[0]));
+            }
+
+            _.each(iouReportObjects, (iouReportObject) => {
+                if (!iouReportObject) {
+                    return;
+                }
+                const iouReportKey = `${ONYXKEYS.COLLECTION.REPORT_IOUS}${iouReportObject.reportID}`;
+                reportIOUData[iouReportKey] = iouReportObject;
+            });
+            return Onyx.mergeCollection(ONYXKEYS.COLLECTION.REPORT_IOUS, {...reportIOUData});
+        })
+        .catch(() => Onyx.merge(ONYXKEYS.IOU, {loading: false, creatingIOUTransaction: false, error: true}))
+        .finally(() => Onyx.merge(ONYXKEYS.IOU, {loading: false, creatingIOUTransaction: false, error: false}));
```

Comment:
I believe finally is still called after catch, so we need to remove error from finally